Fix test suite if mpi4py is not present.

swails opened the pull request:
Title says it all

swails committed:
Fix test suite if mpi4py is not present.

kyleabeauchamp commented:
So we also have a dummy mpi in our repex repo, might be good to synchronize
these
On Apr 29, 2014 9:00 AM, "Jason Swails" notifications@github.com wrote:

> ## Title says it all
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/swails/openmm-tests master
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/choderalab/openmm-tests/pull/2
> Commit Summary
> - Fix test suite if mpi4py is not present.
> 
> File Changes
> - _A_ energy-rms/fake_mpi.pyhttps://github.com/choderalab/openmm-tests/pull/2/files#diff-0(74)
> - _M_ energy-rms/test_energy_rms.pyhttps://github.com/choderalab/openmm-tests/pull/2/files#diff-1(10)
> 
> Patch Links:
> - https://github.com/choderalab/openmm-tests/pull/2.patch
> - https://github.com/choderalab/openmm-tests/pull/2.diff
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/choderalab/openmm-tests/pull/2
> .

swails committed:
Consolidate a bit.  Use dummympi from repex module instead of using my (very
old) version from MMPBSA.py

swails commented:
> So we also have a dummy mpi in our repex repo, might be good to synchronize
> these

I didn't know about that.  I just pulled the fake MPI I put in MMPBSA.py a few years ago since I figured it would fit nicely here.  I updated the repex dummympi code to implement the necessary methods and duck typing that's needed here.  This requires the PR I submitted for the `repex` repo to be updated.

swails committed:
Solve the following exception that occurs when the directory name of the saved
datasets does not yet exist:

Traceback (most recent call last):
  File "test_energy_rms.py", line 318, in <module>
    ncfile = netcdf.Dataset(netcdf_filename, 'w', version='NETCDF4')
  File "netCDF4.pyx", line 1386, in netCDF4.Dataset.__init__ (netCDF4.c:19148)
RuntimeError: Permission denied

also remove redundant line.

swails commented:
Does anybody realistically run `test_energy_rms.py` the way it is now?  It's currently hard-coded to use the `Reference` platform and looks like it'll take over a day to finish just that platform.  The only way I could make progress is to change to the CUDA platform and reduce the number of equilibration steps to 20 from 100.

A couple other questions:
- Is there a reliable way of knowing if the various tests "pass" for those of us who haven't already figured out what the answer _should_ be?
- Is there a reason that only the Reference platform is currently tested?  It looked like the original intent was to test all platforms and all available precision models, but then this was ignored and only the `Reference` platform was tested.
- Maybe only run 10 steps for the Reference platform instead of 100?
